tools/run_tests/report_utils.py

```diff
@@ -69,147 +69,35 @@ def render_xml_report(resultset, xml_report):
   tree.write(xml_report, encoding='UTF-8')
 
 
-# TODO(adelez): Use mako template.
-def fill_one_test_result(shortname, resultset, html_str):
-  if shortname in resultset:
-    # Because interop tests does not have runs_per_test flag, each test is run
-    # once. So there should only be one element for each result.
-    result = resultset[shortname][0] 
-    if result.state == 'PASSED':
-      html_str = '%s<td bgcolor=\"green\">PASS</td>\n' % html_str
-    else:
-      tooltip = ''
-      if result.returncode > 0 or result.message:
-        if result.returncode > 0:
-          tooltip = 'returncode: %d ' % result.returncode
-        if result.message:
-          escaped_msg = _filter_msg(result.message, 'HTML')
-          tooltip = '%smessage: %s' % (tooltip, escaped_msg)       
-      if result.state == 'FAILED':
-        html_str = '%s<td bgcolor=\"red\">' % html_str
-        if tooltip:  
-          html_str = ('%s<a href=\"#\" data-toggle=\"tooltip\" '
-                      'data-placement=\"auto\" title=\"%s\">FAIL</a></td>\n' % 
-                      (html_str, tooltip))
-        else:
-          html_str = '%sFAIL</td>\n' % html_str
-      elif result.state == 'TIMEOUT':
-        html_str = '%s<td bgcolor=\"yellow\">' % html_str
-        if tooltip:
-          html_str = ('%s<a href=\"#\" data-toggle=\"tooltip\" '
-                      'data-placement=\"auto\" title=\"%s\">TIMEOUT</a></td>\n' 
-                      % (html_str, tooltip))
-        else:
-          html_str = '%sTIMEOUT</td>\n' % html_str
-  else:
-    html_str = '%s<td bgcolor=\"magenta\">Not implemented</td>\n' % html_str
-  
-  return html_str
+def render_interop_html_report(
+  client_langs, server_langs, test_cases, auth_test_cases, http2_cases, 
+  resultset, num_failures, cloud_to_prod, http2_interop):
+  """Generate HTML report for interop tests."""
+  from mako.runtime import Context
+  from mako.template import Template
```

Comment:
Seems easy enough to work around:

try:
  from mako.template import Template
  ....
except (ImportError):
  pass # Mako not supported


There are a couple advantages to this over the current:
  - you don't accidentally import the same thing multiple times by calling the function multiple times 
 - you avoid the possibility of another function depending implicitly on these packages based on call order
 - it is explicit that mako is optional, rather than being obscured as it is now.